include/z64schedule.h

```diff
@@ -28,10 +28,15 @@
  */
 
 // Macro to convert the time format used in the save struct into the format used in Schedule
-#define SCHEDULE_CONVERT_TIME(time) ((u16)((time) - 0x10000 / 360 * 90))
+#define SCHEDULE_CONVERT_TIME_ALT(time) ((time) - 0x10000 / 360 * 90)
+#define SCHEDULE_CONVERT_TIME(time) ((u16)(s32)SCHEDULE_CONVERT_TIME_ALT(time))
```

Comment:
I am not 100% sure on having the duplicate macro. We could just have the non cast version, and cast as needed. (I think that is how it was originally, but we never found a no cast instance until now).

Reply:
I'll give this a try and see how it ends up looking

Reply:
It was surprisingly less bad than I expected
https://github.com/zeldaret/mm/pull/1438/commits/1f3269580702cc434ca2e8eb069e256a02780804